### PR TITLE
feature/encapsulate-Tests-field

### PR DESCRIPTION
**Pull request by amibiz**

### CODE REVIEW

**Comment by tebeka** [comment]
Thanks, will take a look soon-ish.

**Comment by tebeka** [comment]
@amibiz The two main things I'd like to get going are #31 and #32 (the latter don't have to be the yacc tool be we need to separate lexer/scanner and "parser" logic)

**Comment by amibiz** [comment]
@tebeka I left a comment on #31. For #32, I would like to "get there" incrementally, not sure how.

**Comment by tebeka** [comment]
OK. Sorry but super busy - will try to have a look soon.

File: suite.go [hunk 1]
```diff
@@ -13,7 +13,46 @@ type Suite struct {
 	Name   string
 	Time   string
 	Status string
-	Tests  []*Test
+	tests  []*Test
+}
+
+// AddTest adds a test case to the suite
+func (suite *Suite) AddTest(test *Test) {
```
Comment (tebeka) [inline suite.go]: I'm note sure I like this change. I don't see value in hiding the data structure implementation when it's internal. I don't think the extra code is worth it and that it's very idiomatic go.

If we'd expose some data structures as an API (see #39 ) then we *might* make `Suite` and `Test` an interface.

Am I missing something?

Comment (amibiz) [inline suite.go]: This change is part of a bigger local refactoring branch. Unfortunately, it has not come to fruition yet, so I thought I will commit and PR early.

**Comment by amibiz** [comment]
As it stands, this PR is not valuable (see @tebeka comment above). Closing it for now.

**Comment by amibiz** [comment]
As it stands, this PR is not valuable (see @tebeka comment above). Closing it for now.

**Pull request closed without merging**